src/vm/codeman.h

```diff
@@ -228,6 +228,43 @@ typedef struct _hpCodeHdr
     }
 #endif // !USE_INDIRECT_CODEHEADER
 
+#if defined(WIN64EXCEPTIONS)
+#ifdef USE_INDIRECT_CODEHEADER
+    typedef struct _hpRealCodeHdr ImplType;
+    typedef PTR_RealCodeHeader    SelfType;
+#else
+    typedef struct _hpCodeHdr     ImplType;
+    typedef PTR_CodeHeader        SelfType;
+#endif
+
+    SelfType Self(void)
+    {
+#ifdef DACCESS_COMPILE
+        return (TADDR) this;
```

Comment:
This cast is not right for `DACCESS_COMPILE`.

The rules for `DACCESS_COMPILE` are explained in https://github.com/dotnet/coreclr/blob/master/src/inc/daccess.h#L138, and the whole story is in https://github.com/dotnet/coreclr/blob/master/Documentation/botr/dac-notes.md.